[Dev-Guide]: Dealing with PR blockers (#712)
ferrocene/specification: src/dev-guide.rst

Reviewer: not sure we should limit discussion to team meets... I think it's fine to just say "bring up the issue to the team"

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/fls/712/3977e597187d55140d57c75227a116b6afc7a852..c69778425b5c44c215fb2cbc7c5dc4df697ecfa3)*

Author: Works for me! Updated.

``<Action>`` must denote either ``Changed``, ``New``, or ``Removed``.

Dealing with PR blockers
------------------------

Incorrect or incomplete semantics
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

While working on a PR, a developer may need to modify existing FLS semantics or terminology, only for the developer to discover that these semantics or terms are either incorrect or incomplete.
In such cases, bring up the issue to the FLS team.

Depending on the scope of the required changes, the FLS team may decide to either:
- Enact the necessary changes in the same PR-in-development, or

> - Approximate the contents of the PR-in-development, and open a GitHub issue to track the update on the semantics or terms.

Reviewer 2: Could we state the condition under which deferring the correction is acceptable? Like we chatted about in our meeting,  in #694 the leading-`::` modeling defect predated the PR and we could keep that PR narrowly scoped without making the defect worse, then track the inherited problem in #711.

As written, "approximate" also appears to permit merging an error introduced by the current PR. I think this option should instead say that a pre-existing, out-of-scope problem may be deferred only when the current PR remains accurate within its stated scope. Otherwise the correction needs to land in the under-review PR or in a prerequisite PR.

I'm a little worried about this being too loose for the conditions under which we can punt to create an issue and address later. Hope you catch my drift.

Author: Reworded.

Missing semantics
~~~~~~~~~~~~~~~~~

> While working on a PR, a developer may need to modify existing FLS semantics or use existing terminology, only for the developer to discover that these semantics or terms are missing.

Reviewer 2: This was kinda hard to parse, as I couldn't tell how to apply it. Seems this branch details the semantics are described as both existing in the FLS and missing from it, sorta.

Could we state the trigger a little more bluntly / directly? For example:

```suggestion
While working on a PR, a developer may discover that the PR depends on semantics or terminology that the FLS does not yet define.
```

This would also distinguish this case from incomplete existing text in the preceding subsection.

Author: Incorporated.
